### ISSUE TITLE
Chaining << into OutputStream doesn't work

### ISSUE THREAD

**Issue by sdcorey** [rubyzip/rubyzip#351]
An illustrative example:

```
Zip::OutputStream.open('file.zip') do |zip|
  zip.put_next_entry(<ValidEntryData>)
  zip << "string1" << "string2" << "string3"
end
```
Only "string1" is properly deflated and written to the output stream.  The remaining items in the chain are written to the output stream undeflated.  This is because OutputStream#<< doesn't return self in order to allow chaining.  Instead it returns @compressor.<<, which in my case resolves to Deflater#<<.  The call to Deflater#<< returns the file stream object, so that any chained calls after the first in the chain end up writing uncompressed data straight to the file stream.

My fix was to override OutputStream::<< by adding the last line, as follows:

```
def << (data)
  @compressor << data
  self
end
```

I would argue that the same should be done for classes Deflater, PassThruCompressor, and maybe for the sake of thoroughness, NullCompressor.

**Comment by simonoff** [comment]
@sdcorey can you make a PR?

**Comment by hainesr** [comment]
I just had a look at the code and `OutputStream#<<` does return self, and has done for 4 years:

```ruby
def <<(data)
  @compressor << data
  self
end
```

(from https://github.com/rubyzip/rubyzip/blob/master/lib/zip/output_stream.rb#L180)

Am I missing something? Happy to try and fix this if I am.

**Comment by hainesr** [comment]
I finally actually tried this and it works, so I think this issue can be closed.

**Comment by jdleesmiller** [comment]
Great, thanks 👍

**Comment by sdcorey** [comment]
Looking back, I see that at the time a gem had pinned my rubyzip to 1.0.0.  The issue was a duplicate of one that had already been closed (#135) by version 1.1.1.  Sorry to waste anyone's time.
